Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -923,24 +923,34 @@ private class MacroApplication<Context: MacroExpansionContext>: SyntaxRewriter {
       return DeclSyntax(node)
     }
 
-    guard node.bindings.count == 1, let binding = node.bindings.first else {
+    guard node.bindings.count == 1 else {
       contextGenerator(Syntax(node)).addDiagnostics(
         from: MacroApplicationError.accessorMacroOnVariableWithMultipleBindings,
         node: node
       )
       return DeclSyntax(node)
     }
-
-    let expansion = expandAccessors(of: node, existingAccessors: binding.accessorBlock)
+    
+    let bindingKeyPath = \VariableDeclSyntax.bindings[node.bindings.startIndex]
```

Comment:
I mildly prefer just keeping `node.bindings[node.bindings.startIndex]` TBH. Maybe we should have a `mutate` method or something similar to make this a little nicer, or just wait until we can use variadic generics and then update `with`...

Reply:
I agree that the bindings key path is a little confusing, it just seems a little over-complicated

What do you think about having `var binding = node.bindings.first` in the `guard`, then using `binding.accessorBlock = …` etc below and finalizing the new node by setting `node.bindings = [binding]`. That seems like the cleanest solution to me.